flow/layers/transform_layer.cc

```diff
@@ -42,12 +41,12 @@ void TransformLayer::Diff(DiffContext* context, const Layer* old_layer) {
 
 void TransformLayer::Preroll(PrerollContext* context) {
   auto mutator = context->state_stack.save();
-  mutator.transform(transform_);
+  mutator.transform(transform_.asM33());
 
   SkRect child_paint_bounds = SkRect::MakeEmpty();
   PrerollChildren(context, &child_paint_bounds);
 
-  transform_.mapRect(&child_paint_bounds);
+  transform_.asM33().mapRect(&child_paint_bounds);
```

Comment:
We should be using a 4x4 mapping function here. Skia has one in SkMatrixPriv because it makes assumptions it doesn't want to advertise as public, but we can live with those assumptions. Just grab it out from there.

Reply:
May be we need to use `extern` to use this symbol because:

```console
The build continued as if that argument was unspecified.

ERROR at //flutter/flow/layers/transform_layer.cc:6:11: Including a private header.
#include "third_party/skia/src/core/SkMatrixPriv.h"
          ^---------------------------------------
This file is private to the target //third_party/skia:skia
```

Reply:
We can't directly use SkMatrixPriv.h because it's private to Skia. But we can write a method that does the same thing (perhaps copying a lot of Skia code to do it).